src/pybind/mgr/dashboard/services/nvmeof_client.py

```diff
@@ -36,6 +36,19 @@ def __init__(self, gw_group: Optional[str] = None):
                     f'Unable to retrieve the gateway info: {e}'
                 )
 
+            # While creating listener need to direct request to the gateway
+            # address where listener is supposed to be added.
+            if traddr:
+                gateways_info = NvmeofGatewaysConfig.get_gateways_config()
+                matched_instance = next(
+                    filter(lambda instance: traddr in instance['service_url'],
+                        (instance for instances in gateways_info['gateways'].values() for instance in instances)),
+                    None
+                )
+                if matched_instance:
+                    self.gateway_addr = matched_instance['service_url']
+                    logger.info(f"Gateway address set to: {self.gateway_addr}")
```

Comment:
This trace could be confusing, it looks like a new GW was set... And I'd rather say [this is debug, not info](https://github.com/ceph/ceph/wiki/Teams-%E2%80%90-Ceph-Dashboard-%E2%80%90-Best-Practices#log-levels).